// adhan/lib/cjs/CalculationParameters.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _Madhab = require("./Madhab.js");
var _HighLatitudeRule = _interopRequireDefault(require("./HighLatitudeRule.js"));
var _PolarCircleResolution = require("./PolarCircleResolution.js");
var _Rounding = require("./Rounding.js");
var _Shafaq = require("./Shafaq.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(a, n) { if (!(a instanceof n)) throw new TypeError("Cannot call a class as a function"); }
function _defineProperties(e, r) { for (var t = 0; t < r.length; t++) { var o = r[t]; o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), Object.defineProperty(e, _toPropertyKey(o.key), o); } }
function _createClass(e, r, t) { return r && _defineProperties(e.prototype, r), t && _defineProperties(e, t), Object.defineProperty(e, "prototype", { writable: !1 }), e; }
function _defineProperty(e, r, t) { return (r = _toPropertyKey(r)) in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
var CalculationParameters = exports["default"] = /*#__PURE__*/function () {
  function CalculationParameters(
  // Name of the method, can be used to apply special behavior in calculations.
  // This property should not be manually modified.
  method) {
    var fajrAngle = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : 0;
    var ishaAngle = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : 0;
    var ishaInterval = arguments.length > 3 && arguments[3] !== undefined ? arguments[3] : 0;
    var maghribAngle = arguments.length > 4 && arguments[4] !== undefined ? arguments[4] : 0;
    _classCallCheck(this, CalculationParameters);
    // Madhab to determine how Asr is calculated.
    _defineProperty(this, "madhab", _Madhab.Madhab.Shafi);
    // Rule to determine the earliest time for Fajr and latest time for Isha
    // needed for high latitude locations where Fajr and Isha may not truly exist
    // or may present a hardship unless bound to a reasonable time.
    _defineProperty(this, "highLatitudeRule", _HighLatitudeRule["default"].MiddleOfTheNight);
    // Manual adjustments (in minutes) to be added to each prayer time.
    _defineProperty(this, "adjustments", {
      fajr: 0,
      sunrise: 0,
      dhuhr: 0,
      asr: 0,
      maghrib: 0,
      isha: 0
    });
    // Adjustments set by a calculation method. This value should not be manually modified.
    _defineProperty(this, "methodAdjustments", {
      fajr: 0,
      sunrise: 0,
      dhuhr: 0,
      asr: 0,
      maghrib: 0,
      isha: 0
    });
    // Rule to determine how to resolve prayer times inside the Polar Circle
    // where daylight or night may persist for more than 24 hours depending
    // on the season
    _defineProperty(this, "polarCircleResolution", _PolarCircleResolution.PolarCircleResolution.Unresolved);
    // How seconds are rounded when calculating prayer times
    _defineProperty(this, "rounding", _Rounding.Rounding.Nearest);
    // Used by the MoonsightingCommittee method to determine how to calculate Isha
    _defineProperty(this, "shafaq", _Shafaq.Shafaq.General);
    this.method = method;
    this.fajrAngle = fajrAngle;
    this.ishaAngle = ishaAngle;
    this.ishaInterval = ishaInterval;
    this.maghribAngle = maghribAngle;
    if (this.method === null) {
      // we don't want a breaking change
      this.method = 'Other';
    }
  }
  return _createClass(CalculationParameters, [{
    key: "nightPortions",
    value: function nightPortions() {
      switch (this.highLatitudeRule) {
        case _HighLatitudeRule["default"].MiddleOfTheNight:
          return {
            fajr: 1 / 2,
            isha: 1 / 2
          };
        case _HighLatitudeRule["default"].SeventhOfTheNight:
          return {
            fajr: 1 / 7,
            isha: 1 / 7
          };
        case _HighLatitudeRule["default"].TwilightAngle:
          return {
            fajr: this.fajrAngle / 60,
            isha: this.ishaAngle / 60
          };
        default:
          throw "Invalid high latitude rule found when attempting to compute night portions: ".concat(this.highLatitudeRule);
      }
    }
  }]);
}();
