// adhan/lib/cjs/PolarCircleResolution.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.polarCircleResolvedValues = exports.PolarCircleResolution = void 0;
var _Coordinates = _interopRequireDefault(require("./Coordinates.js"));
var _SolarTime = _interopRequireDefault(require("./SolarTime.js"));
var _DateUtils = require("./DateUtils.js");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(e, r, t) { return (r = _toPropertyKey(r)) in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
var PolarCircleResolution = exports.PolarCircleResolution = {
  AqrabBalad: 'AqrabBalad',
  AqrabYaum: 'AqrabYaum',
  Unresolved: 'Unresolved'
};
var LATITUDE_VARIATION_STEP = 0.5; // Degrees to add/remove at each resolution step
var UNSAFE_LATITUDE = 65; // Based on https://en.wikipedia.org/wiki/Midnight_sun

var isValidSolarTime = function isValidSolarTime(solarTime) {
  return !isNaN(solarTime.sunrise) && !isNaN(solarTime.sunset);
};
var _aqrabYaumResolver = function aqrabYaumResolver(coordinates, date) {
  var daysAdded = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : 1;
  var direction = arguments.length > 3 && arguments[3] !== undefined ? arguments[3] : 1;
  if (daysAdded > Math.ceil(365 / 2)) {
    return null;
  }
  var testDate = new Date(date.getTime());
  testDate.setDate(testDate.getDate() + direction * daysAdded);
  var tomorrow = (0, _DateUtils.dateByAddingDays)(testDate, 1);
  var solarTime = new _SolarTime["default"](testDate, coordinates);
  var tomorrowSolarTime = new _SolarTime["default"](tomorrow, coordinates);
  if (!isValidSolarTime(solarTime) || !isValidSolarTime(tomorrowSolarTime)) {
    return _aqrabYaumResolver(coordinates, date, daysAdded + (direction > 0 ? 0 : 1), -direction);
  }
  return {
    date: date,
    tomorrow: tomorrow,
    coordinates: coordinates,
    solarTime: solarTime,
    tomorrowSolarTime: tomorrowSolarTime
  };
};
var _aqrabBaladResolver = function aqrabBaladResolver(coordinates, date, latitude) {
  var solarTime = new _SolarTime["default"](date, _objectSpread(_objectSpread({}, coordinates), {}, {
    latitude: latitude
  }));
  var tomorrow = (0, _DateUtils.dateByAddingDays)(date, 1);
  var tomorrowSolarTime = new _SolarTime["default"](tomorrow, _objectSpread(_objectSpread({}, coordinates), {}, {
    latitude: latitude
  }));
  if (!isValidSolarTime(solarTime) || !isValidSolarTime(tomorrowSolarTime)) {
    return Math.abs(latitude) >= UNSAFE_LATITUDE ? _aqrabBaladResolver(coordinates, date, latitude - Math.sign(latitude) * LATITUDE_VARIATION_STEP) : null;
  }
  return {
    date: date,
    tomorrow: tomorrow,
    coordinates: new _Coordinates["default"](latitude, coordinates.longitude),
    solarTime: solarTime,
    tomorrowSolarTime: tomorrowSolarTime
  };
};
var polarCircleResolvedValues = exports.polarCircleResolvedValues = function polarCircleResolvedValues(resolver, date, coordinates) {
  var defaultReturn = {
    date: date,
    tomorrow: (0, _DateUtils.dateByAddingDays)(date, 1),
    coordinates: coordinates,
    solarTime: new _SolarTime["default"](date, coordinates),
    tomorrowSolarTime: new _SolarTime["default"]((0, _DateUtils.dateByAddingDays)(date, 1), coordinates)
  };
  switch (resolver) {
    case PolarCircleResolution.AqrabYaum:
      {
        return _aqrabYaumResolver(coordinates, date) || defaultReturn;
      }
    case PolarCircleResolution.AqrabBalad:
      {
        var latitude = coordinates.latitude;
        return _aqrabBaladResolver(coordinates, date, latitude - Math.sign(latitude) * LATITUDE_VARIATION_STEP) || defaultReturn;
      }
    default:
      {
        return defaultReturn;
      }
  }
};
