// adhan/lib/cjs/SolarTime.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _Astronomical = _interopRequireDefault(require("./Astronomical.js"));
var _MathUtils = require("./MathUtils.js");
var _SolarCoordinates = _interopRequireDefault(require("./SolarCoordinates.js"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(a, n) { if (!(a instanceof n)) throw new TypeError("Cannot call a class as a function"); }
function _defineProperties(e, r) { for (var t = 0; t < r.length; t++) { var o = r[t]; o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), Object.defineProperty(e, _toPropertyKey(o.key), o); } }
function _createClass(e, r, t) { return r && _defineProperties(e.prototype, r), t && _defineProperties(e, t), Object.defineProperty(e, "prototype", { writable: !1 }), e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
var SolarTime = exports["default"] = /*#__PURE__*/function () {
  function SolarTime(date, coordinates) {
    _classCallCheck(this, SolarTime);
    var julianDay = _Astronomical["default"].julianDay(date.getFullYear(), date.getMonth() + 1, date.getDate(), 0);
    this.observer = coordinates;
    this.solar = new _SolarCoordinates["default"](julianDay);
    this.prevSolar = new _SolarCoordinates["default"](julianDay - 1);
    this.nextSolar = new _SolarCoordinates["default"](julianDay + 1);
    var m0 = _Astronomical["default"].approximateTransit(coordinates.longitude, this.solar.apparentSiderealTime, this.solar.rightAscension);
    var solarAltitude = -50.0 / 60.0;
    this.approxTransit = m0;
    this.transit = _Astronomical["default"].correctedTransit(m0, coordinates.longitude, this.solar.apparentSiderealTime, this.solar.rightAscension, this.prevSolar.rightAscension, this.nextSolar.rightAscension);
    this.sunrise = _Astronomical["default"].correctedHourAngle(m0, solarAltitude, coordinates, false, this.solar.apparentSiderealTime, this.solar.rightAscension, this.prevSolar.rightAscension, this.nextSolar.rightAscension, this.solar.declination, this.prevSolar.declination, this.nextSolar.declination);
    this.sunset = _Astronomical["default"].correctedHourAngle(m0, solarAltitude, coordinates, true, this.solar.apparentSiderealTime, this.solar.rightAscension, this.prevSolar.rightAscension, this.nextSolar.rightAscension, this.solar.declination, this.prevSolar.declination, this.nextSolar.declination);
  }
  return _createClass(SolarTime, [{
    key: "hourAngle",
    value: function hourAngle(angle, afterTransit) {
      return _Astronomical["default"].correctedHourAngle(this.approxTransit, angle, this.observer, afterTransit, this.solar.apparentSiderealTime, this.solar.rightAscension, this.prevSolar.rightAscension, this.nextSolar.rightAscension, this.solar.declination, this.prevSolar.declination, this.nextSolar.declination);
    }
  }, {
    key: "afternoon",
    value: function afternoon(shadowLength) {
      // TODO source shadow angle calculation
      var tangent = Math.abs(this.observer.latitude - this.solar.declination);
      var inverse = shadowLength + Math.tan((0, _MathUtils.degreesToRadians)(tangent));
      var angle = (0, _MathUtils.radiansToDegrees)(Math.atan(1.0 / inverse));
      return this.hourAngle(angle, true);
    }
  }]);
}();
